lib/nxdk/net.c

```diff
@@ -152,12 +151,14 @@ int nxNetInit(const nx_net_parameters_t *parameters)
                                  (parameters->ipv4_dns2 >> 24 & 0xff));
             dns[0].type = IPADDR_TYPE_V4;
             dns[1].type = IPADDR_TYPE_V4;
-            dns_override = true;
         }
     }
 
-    if (dns_override) {
-        dns_setserver(2, dns);
+    if (dns[0].u_addr.ip4.addr != 0) {
+        dns_setserver(0, &dns[0]);
+    }
```

Comment:
What if the user had previously set a DNS server, and now wants to deactivate it by setting it to all-zero? Could that be a realistic use-case?

Also affects `(parameters->ipv4_dns1 != 0 || parameters->ipv4_dns2 != 0)` check above.

Reply:
If the user wants to change the DNS settings it's best to do that via the config sector functions. The idea behind allowing overriding here is to allow applications to force lwIP to use a specific DNS server, no matter what the Xbox was configure for.

We could extend this API in the future to allow changing settings after initialization, then a program could dynamically remove or add DNS servers (now that I think about it, this might be what you meant? Initializing with custom DNS server IPs, and then removing them later, falling back to the settings from the config sector?). We'll probably want to keep a copy of the settings or re-read the settings from the config sector in those functions.

Reply:
Is this in the best interest of the user? I'd be quite annoyed if I found an application not using the servers I set. Such an application could have malicious intent and allow for rebinding.

Reply:
There is nothing that nxdk can or could do to keep developers from implementing malicious behavior, and an application not using your preferred DNS server will be the least of your worries in case you run malware.
That said, allowing software to use the DNS server of their choice is common, and there are legitimate usage scenarios (such as testing for DNS hijacking). Not to mention that this feature is always there, this PR is just about fixing a bug in our wrapping API.